### PR TITLE
#161983635 Implement get a single user profile feature

### PR DESCRIPTION
**Pull request by Darthrighteous**
#### What does this PR do?
Build the route to get a single user's profile

#### Description of Task to be completed?
- add endpoint to get a user's profile
- ensure user profile does not return sensitive information when being requested by another user

#### How should this be manually tested?
- clone the repository, cd into it, checkout this branch `feature/161983635/get-single-user-profile`
- set up your .env variables and local database
- run `npm install` to install required packages
- run `npm start` to start up the server
- use postman(or any api testing tool) to test the following routes route
- create a user using the route `POST http://localhost:3000/api/v1/users/signup` providing necessary request body.
- get a user profile with the route `POST http://localhost:3000/api/v1/users/:userId` providing the token in the `Authorization` header as shown in screen shots below.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[161983635](https://www.pivotaltracker.com/story/show/161983635)

#### Screenshots (if appropriate)
My profile:
screen shot 2018-11-19 at 11 08 56 am

Other profile:
screen shot 2018-11-16 at 9 16 58 pm

#### Questions:
N/A

### CODE REVIEW

File: server/controllers/UsersController.js [hunk 1]
```diff
@@ -269,6 +269,46 @@ class UsersController {
...
+
+    // check if the requester is not the owner of the user profile
+    if (requesterId !== +userId) {
+      // delete sensitive information
+      delete userProfile.email;
+      delete userProfile.password;
+      delete userProfile.updatedAt;
+      delete userProfile.facebookId;
+      delete userProfile.twitterId;
+      delete userProfile.googleId;
```
Comment (sulenchy) [inline server/controllers/UsersController.js]: Good job for protecting the users from potential attacks. Keep it up

**Comment by mrSerious** [comment]
LGTM

**Comment by KvNGCzA** [comment]
LGTM

File: server/controllers/UsersController.js [hunk 2]
```diff
@@ -269,6 +269,46 @@ class UsersController {
...
+      });
+    } catch (err) {
+      next(err);
+    }
+
+    // check if the requester is not the owner of the user profile
+    if (requesterId !== +userId) {
+      // delete sensitive information
+      delete userProfile.email;
+      delete userProfile.password;
```
Comment (mbilesanmi) [inline server/controllers/UsersController.js]: All of these except `email` and `updatedAt` should not be shown to the user. What will he use his fbId, twitterId for?

Comment (mbilesanmi) [reply server/controllers/UsersController.js]: What will the user do with his password on the frontend? What will you need the fb, twitter or google IDs for on the front-end? @Darthrighteous

Comment (Darthrighteous) [reply server/controllers/UsersController.js]: Yeah the password field has even been removed with the new implementation we adopted. I'll make these updates.

Comment (Darthrighteous) [reply server/controllers/UsersController.js]: @mbilesanmi fixed!

**Pull request merged**